File: tests/fixtures/v2-outbound-flow/positive/OutboundFlow.cs
using System.Net.Http;

class OutboundFlow
{
    object Direct(dynamic Request, HttpClient client)
    {
        return client.GetAsync(Request.Query["direct"]);
    }

    object Propagated(dynamic Request, HttpClient client)
    {
        var requested = Request.Query["propagated"];
        var forwardedValue = requested;
        return client.GetStringAsync(forwardedValue);
    }

    object Parsed(dynamic Request, HttpClient client)
    {
        var parsed = new Uri(Request.Query["parsed"]);
        return client.GetAsync(parsed);
    }

    bool ValidScheme(Uri parsed)
    {
        return parsed.Scheme == Uri.UriSchemeHttps;
    }
}

File: tests/fixtures/v2-outbound-flow/positive/OutboundFlow.java
import java.net.URI;
import java.net.http.HttpRequest;

class OutboundFlow {
    Object direct(HttpServletRequest request) {
        return HttpRequest.newBuilder(request.getParameter("direct"));
    }

    Object propagated(HttpServletRequest request) {
        String requested = request.getParameter("propagated");
        String alias = requested;
        return HttpRequest.newBuilder(alias);
    }

    Object parsed(HttpServletRequest request) {
        URI parsed = URI.create(request.getParameter("parsed"));
        return HttpRequest.newBuilder(parsed);
    }

    boolean validScheme(URI parsed) {
        return parsed.getScheme().equals("https");
    }
}
